evalml/pipelines/components/transformers/imputers/imputer.py

```diff
@@ -125,29 +124,24 @@ def fit(self, X, y=None):
         )
         numeric_cols = list(X.ww.select(["numeric"], return_schema=True).columns)
 
-        # TODO: Remove this when columns with True/False/NaN are inferred properly as BooleanNullable.
-        # If columns with boolean values and NaN are included with normal categorical columns, columns
-        # with object dtypes are attempted to be cast to float64 with scikit-learn 1.1.  So we separate
-        # boolean and categorical into separate imputers.
-        for col in cat_cols:
-            if is_categorical_actually_boolean(X, col):
-                cat_cols.remove(col)
-                bool_cols.append(col)
-
         nan_ratio = X.isna().sum() / X.shape[0]
         self._all_null_cols = nan_ratio[nan_ratio == 1].index.tolist()
 
-        X_numerics = X[[col for col in numeric_cols if col not in self._all_null_cols]]
+        X_numerics = X.ww[
+            [col for col in numeric_cols if col not in self._all_null_cols]
+        ]
         if len(X_numerics.columns) > 0:
             self._numeric_imputer.fit(X_numerics, y)
             self._numeric_cols = X_numerics.columns
 
-        X_categorical = X[[col for col in cat_cols if col not in self._all_null_cols]]
+        X_categorical = X.ww[
+            [col for col in cat_cols if col not in self._all_null_cols]
+        ]
         if len(X_categorical.columns) > 0:
             self._categorical_imputer.fit(X_categorical, y)
             self._categorical_cols = X_categorical.columns
 
-        X_boolean = X[[col for col in bool_cols if col not in self._all_null_cols]]
+        X_boolean = X.ww[[col for col in bool_cols if col not in self._all_null_cols]]
```

Comment:
When using the imputer in pipelines, we weren't fitting the child imputers on woodwork-initialized data.

I'm surprised this hasn't been noticed before, but I stumbled upon it because of these changes. Now that we're back to returning early if all the columns are the `bool` dtype, this was causing an issue in a `BooleanNullable` column that would get re-inferred as `Boolean` at fit, so we'd return early and then raise an error that the `_component_obj` wasn't fit at transform when the correct `BooleanNullable` type was passed in.